src/components/app/app.js

```diff
@@ -1,6 +1,24 @@
 import angular from "angular";
 import template from "./app.html";
 
-export const appModule = angular.module("appModule", []).component("app", {
-  template
-}).name;
+import ngMaterial from "angular-material";
+import ngAnimate from "angular-animate";
+import ngAria from "angular-aria";
+
+import { transactionDialogModule } from "../transaction-dialog/transaction-dialog";
+
+export class AppController {
+  constructor() {
+    this.transactionDialogOpen = { isOpen: false };
+  }
+  handleAddTransactionClick(event) {
+    this.transactionDialogOpen = { isOpen: true, openEvent: event };
```

Comment:
why do you need to group these things into one object? you can pass it as separate 2 props to component